Plugins/NativeEngine/CMakeLists.txt

```diff
@@ -52,6 +52,8 @@ endif()
 
 target_compile_definitions(NativeEngine
     PRIVATE NOMINMAX)
+target_compile_definitions(NativeEngine
+    PRIVATE API${GRAPHICS_API}) # OpenGL is defined in bgfx.h. Using APIXXX instead
```

Comment:
I think this should probably be in all CAPS to match typical macros which may avoid the conflict. Use https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#string-transformations to convert to uppercase. But ideally, this shouldn't be done at all. The specific code should be in the specific files instead of using macros.

Reply:
I agree we should not see any `#ifdef platform` anywhere in the code but in specific .cpp files. It's mostly in shaderCompilerCommon.cpp so I'll let you do the change when you are done with your shader compilation work.